### ISSUE TITLE
Error: The 'cache' field on 'RequestInitializerDict' is not implemented.

### ISSUE THREAD

**Issue by r3dpoint** [smithy-lang/smithy-typescript#1407]
Started seeing this error when using AWS S3 SDK to GetObject in a Cloudflare Pages App, which runs on workerd:

```
✘ [ERROR] Error: The 'cache' field on 'RequestInitializerDict' is not implemented.

 at FetchHttpHandler.handle
  (file:///.../node_modules/@smithy/fetch-http-handler/dist-es/fetch-http-handler.js:77:30)
      at null.<anonymous> (async
```

I will attempt a downgrade. I may try a standalone app to share.

Update pinning to 3.2.5 resolves the error.

**Comment by nicksrandall** [comment]
Also running into this... cache setting is not implemented in Cloudflare workers.

**Comment by flexchar** [comment]
Same, cloudflare workerd runtime.

Had to downgrade to `3.645.0`

```json
"@aws-sdk/client-s3": "3.645.0",
"@aws-sdk/s3-request-presigner": "3.645.0",
```

**Comment by nicksrandall** [comment]
I believe the error is here: https://github.com/smithy-lang/smithy-typescript/blob/main/packages/fetch-http-handler/src/fetch-http-handler.ts#L115

This is the breaking commit: https://github.com/smithy-lang/smithy-typescript/commit/cf9257e5c37a2eb12d8df257e441b9bf7cc51158

**Comment by kuhe** [comment]
https://github.com/cloudflare/workerd/issues/698

**Comment by kuhe** [comment]
We will accept the patch once approved, but until then there is a workaround posted in the linked issue

```ts
const client = new SDKClient({
  requestHandler: {
    // Cloudflare doesn't support the cache option, so we remove it.
    requestInit: () => ({
      cache: undefined,
    }),
  },
});
```

**Comment by kuhe** [comment]
released in https://www.npmjs.com/package/@smithy/fetch-http-handler/v/3.2.8

**Comment by r3dpoint** [comment]
confirmed fixed for me with latest aws client s3 + v3.2.8 of fetch-http-handler
